python/ray/ml/train/integrations/huggingface/huggingface_trainer.py

```diff
@@ -0,0 +1,190 @@
+                scaling_config=scaling_config,
+                datasets={"train": ray_train_ds, "validation": ray_validation_ds},
+            )
+            result = trainer.fit()
+
+    Args:
+        trainer_init_per_worker: The function that returns an instantiated
+            ``transformers.Trainer`` object and takes in the following arguments:
+            train datset, optional evaluation datset, and config as kwargs.
+        trainer_init_config: Configurations to pass into
+            ``trainer_init_per_worker`` as kwargs.
+        torch_config: Configuration for setting up the PyTorch backend. If set to
+            None, use the default configuration. This replaces the ``backend_config``
+            arg of ``DataParallelTrainer``. Same as in ``TorchTrainer``.
+        scaling_config: Configuration for how to scale data parallel training.
+        run_config: Configuration for the execution of the training run.
+        datasets: Any Ray Datasets to use for training. Use
+            the key "train" to denote which dataset is the training
+            dataset and (optionally) key "validation" to denote the validation
+            dataset. Can only contain a training dataset
+            and up to one extra dataset to be used for validation.
+            If a ``preprocessor`` is provided and has not already been fit,
+            it will be fit on the training dataset. All datasets will be
+            transformed by the ``preprocessor`` if one is provided.
+        preprocessor: A ray.ml.preprocessor.Preprocessor to preprocess the
+            provided datasets.
+        resume_from_checkpoint: A checkpoint to resume training from.
+    """
+
+    def __init__(
+        self,
+        trainer_init_per_worker: Callable[[Dataset, Optional[Dataset], Any], Trainer],
+        trainer_init_config: Optional[Dict] = None,
+        torch_config: Optional[TorchConfig] = None,
+        scaling_config: Optional[ScalingConfig] = None,
+        run_config: Optional[RunConfig] = None,
+        datasets: Optional[Dict[str, GenDataset]] = None,
+        preprocessor: Optional[Preprocessor] = None,
+        resume_from_checkpoint: Optional[Checkpoint] = None,
+    ):
+
+        self._validate_train_loop_per_worker(
+            trainer_init_per_worker, "trainer_init_per_worker"
+        )
+
+        assert TRAIN_DATASET_KEY in datasets
+        assert all(
+            key in (TRAIN_DATASET_KEY, VALIDATION_DATASET_KEY) for key in datasets
+        )
+
+        super().__init__(
+            self._create_train_func(trainer_init_per_worker),
+            trainer_init_config,
+            torch_config,
+            scaling_config,
+            run_config,
+            datasets,
+            preprocessor,
+            resume_from_checkpoint,
+        )
+
+    def _create_train_func(self, trainer_init_per_worker):
+        def train_loop_per_worker(config):
+            train_dataset = None
+            eval_dataset = None
```

Comment:
how do these 2 work? always hardcoded to be None?

Reply:
It's just temporary to allow for lint to pass on this PR. Will be actually filled out during implementation

Reply:
Can you just do "# noqa" instead?
At the very least, please add comments for these 2 lines.